docs/guide/custom-functions.md

```diff
@@ -262,6 +262,11 @@ declare a companion method for the array size. To do that, provide the
 function arguments and returns an instance of the
 [`ArraySize` class](../api/classes/arraysize.md).
 
+::: tip
+The `arraySizeMethod` is called everytime the cell formula changes, but not when its dependencies change.
+This may cause unexpected behavior when the size of the result array depends on the values of the cells referenced in the formula.
+:::
```

Comment:
@kirszenbaum Do you think it is better to keep it here or in the `known-limitations` guide?

Reply:
I would do both and link between them: `custom-functions` should include this information because it's important to implementing a custom range function. At the same time, `known-limitations` should list all known limitations.

In `custom-functions`, I'd add sth like (If I understand it correctly):

> When you use your custom function in a formula, `arraySizeMethod` is triggered every time the formula changes, but not when the dependencies of the formula change. This can cause unexpected results if the size of the result array depends on the values in the referenced cells.

I'd also change "A function may return multiple values in the form of an array." to "A function **can** return multiple values in the form of an array."

In `known-limitations` (perhaps in a separate "Custom functions" section?), I'd add sth like:

> Custom functions don't automatically recalculate the size of the result arrays when formula dependencies change.

WDYT?

Reply:
I changed it. WHYT now?